### PR TITLE
Home work 15

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/Sendable.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+public interface Sendable<T> {
+	public String getTo();
+
+	public T getContent();
+}
```
Comment (Sofasmile) [inline src/Sendable.java]: add a blank line after }

File: src/MailMessage.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+public class MailMessage implements Sendable<String> {
+	private String sFrom;
```
Comment (Sofasmile) [inline src/MailMessage.java]: try to use more understandable name. maybe better 'from'?

File: src/MailMessage.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	private String sTo;
+	private String sContent;
+
+	public String getFrom() {
```
Comment (Sofasmile) [inline src/MailMessage.java:13]: better add method getFrom to interface

File: src/MailMessage.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	public String getTo() {
+		return sTo;
+	}
+
+	@Override
+	public String getContent() {
+		return sContent;
+	}
+
+	public MailMessage(String From, String To, String Content) {
```
Comment (Sofasmile) [inline src/MailMessage.java]: https://mate-academy.github.io/style-guides/java/java.html#s3.4.2-ordering-class-contents

File: src/MailMessage.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+		sFrom = From;
+		sTo = To;
+		sContent = Content;
+	}
+}
```
Comment (Sofasmile) [inline src/MailMessage.java]: add a blank line

File: src/Salary.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+public class Salary implements Sendable<Integer> {
+	private String sJob;
+	private String sName;
+	private Integer iSalary;
```
Comment (Sofasmile) [inline src/Salary.java]: the same about naming

File: src/MailService.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class MailService<T> implements Consumer<Sendable<T>> {
+	private final Map<String, List<T>> msMailBox;
```
Comment (Sofasmile) [inline src/MailService.java]: do we need final here?

File: src/MailService.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+	public MailService() {
+		msMailBox = new HashMap<String, List<T>>() {
+			@Override
+			@SuppressWarnings("empty-statement")
+			public List<T> get(Object key) {
+				if (msMailBox.containsKey(key)) {
```
Comment (Sofasmile) [inline src/MailService.java]: we can remove if-else statement form here 
and use method getOrDefault()

File: src/MailService.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+					return msMailBox.getOrDefault(key, null);
+				} else {
+					List<T> lst = new LinkedList<>();
+					msMailBox.put((String) key, lst);
+					return lst;
+
```
Comment (Sofasmile) [inline src/MailService.java]: remove redundant lines

Comment (UnitOmega13) [reply src/MailService.java]: All fixed according to commits

File: src/MailMessage.java (hunk 5 above)
Comment (Sofasmile) [reply src/MailMessage.java]: add a blank line after }

File: src/MailService.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+public class MailService<T> implements Consumer<Sendable<T>> {
+	private Map<String, List<T>> msMailBox;
+
+	public MailService() {
+		msMailBox = new HashMap<String, List<T>>() {
+			@Override
+			@SuppressWarnings("empty-statement")
+			public List<T> get(Object key) {
+				if (!msMailBox.containsKey(key)) {
```
Comment (Sofasmile) [inline src/MailService.java]: if (!msMailBox.containsKey(key)) {
					List<T> lst = new LinkedList<>();
					msMailBox.put((String) key, lst);
					return lst;
				}
				return msMailBox.getOrDefault(key, null);
---->
return msMailBox.put((String) key, new LinkedList<>());

File: src/MailService.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class MailService<T> implements Consumer<Sendable<T>> {
+	private Map<String, List<T>> msMailBox;
```
Comment (Sofasmile) [inline src/MailService.java]: what does mean 'ms' from msMailBox? maybe better rename?

File: src/MailMessage.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+	public String getTo() {
+		return destination;
+	}
+
+	@Override
+	public String getContent() {
+		return packageType;
+	}
+
+	public MailMessage(String sender, String destination, String packageType) {
```
Comment (Sofasmile) [inline src/MailMessage.java]: we need create constructor after fields and before methods in the class

File: src/Salary.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
+public class Salary implements Sendable<Integer> {
+	private String job;
+	private String name;
+	private Integer salary;
+
+	public Salary(String job, String name, Integer salary) {
+		job = job;
```
Comment (Sofasmile) [inline src/Salary.java]:
```suggestion
		this.job = job;
```

Comment (UnitOmega13) [reply src/Salary.java]: Changed

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**